llvm-project/Dockerfile.epoch3

```diff
@@ -0,0 +1,90 @@
+ARG base=ghcr.io/clangbuiltlinux/llvm-project:stage2
+
+FROM alpine:edge AS source
+RUN wget --no-verbose https://git.kernel.org/torvalds/t/linux-5.18-rc6.tar.gz
+RUN wget --no-verbose https://musl.libc.org/releases/musl-1.2.3.tar.gz
+RUN wget --no-verbose https://zlib.net/zlib-1.2.12.tar.gz
+RUN wget --no-verbose https://github.com/llvm/llvm-project/releases/download/llvmorg-14.0.1/llvm-project-14.0.1.src.tar.xz
+
+FROM ${base} AS stage2
+FROM alpine:edge AS stage3
+
+### BEGIN STAGE3
+
+COPY --from=stage2 /usr/local/bin /usr/local/bin
+COPY --from=stage2 /usr/local/lib /usr/local/lib
+COPY --from=stage2 /usr/local/include /usr/local/include
+RUN cd /usr/lib/ && \
+  for library in libc++abi.so.1 libc++.a libc++abi.a libc++.so.1 libunwind.so.1 libunwind.a; \
+    do ln -s "/usr/local/lib/x86_64-alpine-linux-musl/${library}" . ; \
+  done
+
+### Linux
+COPY --from=source linux-5.18-rc6.tar.gz .
+RUN tar xf linux-5.18-rc6.tar.gz
+RUN apk add make musl-dev rsync
+RUN make -C linux-5.18-rc6 INSTALL_HDR_PATH=/sysroot/usr LLVM=1 -j$(nproc) headers_install
+RUN apk del rsync musl-dev make
+
+### Musl
+COPY --from=source musl-1.2.3.tar.gz .
+RUN tar xf musl-1.2.3.tar.gz
+ARG MUSL_DIR=musl-1.2.3/build
+RUN mkdir -p ${MUSL_DIR}
+RUN cd ${MUSL_DIR} && \
+  CC=clang AR=llvm-ar RANLIB=llvm-ranlib \
+  ../configure --prefix=/usr --syslibdir=/usr/lib
```

Comment:
something is super fishy to me here...musl doesn't seem to depend on the kernel headers for a build.  They were installed into `/sysroot/usr`, which we never told musl about.  That doesn't seem right to me, but perhaps musl has its own syscall wrappers?

Reply:
https://git.musl-libc.org/cgit/musl/tree/INSTALL#n30

Reply:
I guess https://musl.libc.org/doc/1.1.24/manual.md says "The standard headers are fully self-contained, and do not make use of kernel-provided or compiler-provided headers or otherwise require such headers to be present."